src/detection/scripts/ship_generator.py

```diff
@@ -0,0 +1,334 @@
+from os.path import join
+from random import randrange
+import argparse
+
+import numpy as np
+import pandas as pd
+import rasterio
+import json
+from shapely import geometry
+from shapely.ops import transform
+import pyproj
+from functools import partial
+import matplotlib.pyplot as plt
+
+from data_extent import data, folders
+
+
+def visualize(bbox_coords, windows, culled_windows, src_ds):
+    x, y, x1, y1, x2, y2 = [], [], [], [], [], []
+    full = []
+    for r in windows:
+        full.append([r[0], r[1], r[2], r[3]])
+        full.append([r[0], r[1], r[3], r[2]])
+    valid = []
+    for r in culled_windows:
+        valid.append([r[0], r[2], r[3], r[1]])
+        valid.append([r[0], r[2], r[1], r[3]])
+    for bbox in bbox_coords:
+            x.append(bbox[2])
+            y.append(bbox[0])
+    for window in valid:
+        x1.append(window[2])
+        x1.append(window[3])
+        y1.append(window[0])
+        y1.append(window[1])
+    for window in full:
+        x2.append(window[2])
+        x2.append(window[3])
+        y2.append(window[0])
+        y2.append(window[1])
+
+    image = src_ds.read()
+    image = np.transpose(image, [1, 2, 0])
+
+    plt.imshow(image, origin='upper')
+    plt.scatter(x, y, color='red')
+    plt.scatter(x1, y1, color='blue')
+    plt.scatter(x2, y2, color='green')
+    plt.show()
+
+
+def get_image_corners(x):
+    return [
+        x[0][0],  # minX
+        x[2][0],
+        x[1][1],  # minY
+        x[0][1]
+    ]
+
+
+def coordinate_width_and_height(corners):
+    return corners[1]-corners[0], corners[3]-corners[2]
+
+
+def points_from_rectangle(rectangle, maxXcoord, minYcoord):
+    rectangle_js = rectangle['geometry']['coordinates'][0]
```

Comment:
Oh, I think it's `json`. Instead of this (which you do in some other functions), just pass in the minimal structure you need (what you are now calling `rectangle_js`) and call it `rectangle`.